src/test/kotlin/study/StringCalculator.kt

```diff
@@ -0,0 +1,72 @@
+package study
+
+class StringCalculator {
+    fun calculate(arithmeticExpression: String): Float {
+        if (arithmeticExpression.isBlank()) {
+            throw IllegalArgumentException("문자열이 비어있습니다.")
+        }
+
+        val splitedExpression = splitByBlank(arithmeticExpression)
+        val numbers = splitedExpression.filter { isNumber(it) }
+        val arithmeticSymbols = splitedExpression.filter { isOperator(it) }
+
+        validate(splitedExpression, numbers, arithmeticSymbols)
+
+        val result = numbers.map { it.toFloat() }
+            .toTypedArray()
+            .reduceIndexed { index, acc, s ->
+                doCalculate(acc, s, arithmeticSymbols[index - 1])
+            }
+
+        println(result)
+        return result
+    }
+
+    private fun validate(splitedExpression: List<String>, numbers: List<String>, arithmeticSymbols: List<String>) {
+        if (splitedExpression.size != numbers.size + arithmeticSymbols.size) {
+            throw IllegalArgumentException("숫자와 기호외의 문자가 포함되었습니다.")
+        }
+
+        if (numbers.size != arithmeticSymbols.size + 1) {
+            throw IllegalArgumentException("숫자는 기호보다 1개 많아야 합니다.")
+        }
+
+
+    }
+
+    private fun splitByBlank(arithmeticExpression: String): List<String> {
+        return arithmeticExpression.split(Regex(" +"))
+    }
+
+    private fun isNumber(input: String): Boolean {
+        return Regex("\\d+(\\.\\d+)?").matches(input)
+    }
+
+    private fun isOperator(input: String): Boolean {
+        return input == "+" || input == "-" || input == "*" || input == "/"
```

Comment:
코틀린은 `in`이라는것을 통해 쉽게 해결할 수 있습니다!
```suggestion
        return input in listOf("+" ,"-", "*", "/")
```